File: Java/algorithms/searching/BinarySearch.java
package Java.algorithms.searching;
public class BinarySearch {

    public static void main(String[] args) {
        int[] arr = {2,3,5,7,9,12,15};
        int target = 4;
//        int result = search(arr, target);
        int result = recursiveSearch(arr, target,0 , arr.length-1);
        if(result == -1) System.out.println("Number is not found in the array");
        else System.out.println("Number is present at " + result + " index" );
    }
    public static int search(int[] arr, int target){
        int start = 0;
        int end = arr.length - 1;
        while (start <= end){
            int mid = start + ((end - start)/2);

            if(arr[mid] == target){
                return mid;
            }

            if(arr[mid] > target){
                end = mid-1;
            }
            else start = mid+1;
        }

        return -1;
    }

    //Using recursion
    public static int recursiveSearch(int[] arr, int target, int start, int end){

        if(start <= end){
            int mid = start + ((end - start)/2);

            if(arr[mid] == target){
                return mid;
            }

            if(arr[mid] > target){
                return recursiveSearch(arr, target, start, mid-1);
            }
            else return recursiveSearch(arr, target, mid+1, end);
        }

        return -1;
    }
}

File: Javascript/algorithms/searching/BinarySearch.js
const BinarySearch = (arr, target) => {
  let start = 0
  let end = arr.length - 1

  while (start <= end) {
    let mid = Math.floor((start + end) / 2)

    if (arr[mid] === target) return mid
    else if (arr[mid] > target) end = mid - 1
    else if (arr[mid] < target) start = mid + 1
  }

  return -1
}

console.log(BinarySearch([1, 2, 4, 5, 6, 7, 11, 24], 0))
